modules/client/src/types.ts

```diff
@@ -420,60 +418,96 @@ export type DecodedInitializeFromParams = {
+  INVALID_GRANT_UPDATE_PERMISSION_WHO_ADDRESS =
+    "invalidGrantUpdatePermissionWhoAddress",
+  INVALID_GRANT_UPDATE_PERMISSION_WHERE_ADDRESS =
+    "invalidGrantUpdatePermissionWhereAddress",
+  INVALID_GRANT_UPDATE_PERMISSION_PERMISSION =
+    "invalidGrantUpdatePermissionPermission",
+  INVALID_GRANT_UPDATE_PERMISSION_VALUE =
+    "invalidGrantUpdatePermissionCondition",
+  INVALID_GRANT_UPDATE_PERMISSION_PERMISSION_ID =
+    "invalidGrantUpdatePermissionPermissionId",
+  // Revoke UPDATE_PLUGIN_PERMISSION action
+  INVALID_REVOKE_UPDATE_PERMISSION_WHO_ADDRESS =
+    "invalidRevokeUpdatePermissionWhoAddress",
+  INVALID_REVOKE_UPDATE_PERMISSION_WHERE_ADDRESS =
+    "invalidRevokeUpdatePermissionWhereAddress",
+  INVALID_REVOKE_UPDATE_PERMISSION_PERMISSION =
+    "invalidRevokeUpdatePermissionPermission",
+  INVALID_REVOKE_UPDATE_PERMISSION_VALUE =
+    "invalidRevokeUpdatePermissionCondition",
+  INVALID_REVOKE_UPDATE_PERMISSION_PERMISSION_ID =
+    "invalidRevokeUpdatePermissionPermissionId",
+  // Grant ROOT_PERMISSION action
+  INVALID_GRANT_ROOT_PERMISSION_WHO_ADDRESS =
+    "invalidGrantRootPermissionWhoAddress",
+  INVALID_GRANT_ROOT_PERMISSION_WHERE_ADDRESS =
+    "invalidGrantRootPermissionWhereAddress",
+  INVALID_GRANT_ROOT_PERMISSION_PERMISSION =
+    "invalidGrantRootPermissionPermission",
+  NON_ZERO_GRANT_ROOT_PERMISSION_CALL_VALUE = "nonZeroGrantRootPermissionCallValue",
+  INVALID_GRANT_ROOT_PERMISSION_PERMISSION_ID =
+    "invalidGrantRootPermissionPermissionId",
+  // Revoke ROOT_PERMISSION action
+  INVALID_REVOKE_ROOT_PERMISSION_WHO_ADDRESS =
+    "invalidRevokeRootPermissionWhoAddress",
+  INVALID_REVOKE_ROOT_PERMISSION_WHERE_ADDRESS =
+    "invalidRevokeRootPermissionWhereAddress",
+  INVALID_REVOKE_ROOT_PERMISSION_PERMISSION =
+    "invalidRevokeRootPermissionPermission",
+  INVALID_REVOKE_ROOT_PERMISSION_VALUE = "invalidRevokeRootPermissionCondition",
+  INVALID_REVOKE_ROOT_PERMISSION_PERMISSION_ID =
+    "invalidRevokeRootPermissionPermissionId",
+  // applyUpdate action
+  INVALID_APPLY_UPDATE_ACTION_VALUE = "invalidApplyUpdateActionValue",
   PLUGIN_NOT_INSTALLED = "pluginNotInstalled",
   NOT_ARAGON_PLUGIN_REPO = "notAragonPluginRepo",
   MISSING_PLUGIN_REPO = "missingPluginRepo",
   MISSING_PLUGIN_PREPARATION = "missingPluginPreparation",
-  INVALID_ALLOW_FAILURE_MAP = "invalidAllowFailureMap",
-  INVALID_PLUGIN_RELEASE = "invalidPluginRelease",
-  INVALID_PLUGIN_BUILD = "invalidPluginBuild",
+  UPDATE_TO_INCOMPATIBLE_RELEASE = "updateToIncompatibleRelease",
+  UPDATE_TO_OLDER_OR_SAME_BUILD = "updateToOlderOrSameBuild",
   INVALID_DATA = "invalidData",
   INVALID_PLUGIN_REPO_METADATA = "invalidPluginRepoMetadata",
 }
 
-export type IsPluginUpdateProposalValidParams = {
-  proposalId: string;
-  version: VersionTag;
-  pluginAddress: string;
-  pluginPreparationIndex?: number;
-};
-
 export enum DaoUpdateProposalInvalidityCause {
+  PROPOSAL_NOT_FOUND = "proposalNotFound",
   INVALID_ACTIONS = "invalidActions",
```

Comment:
Is this a duplicate of `INVALID_ACTIONS` in `export enum ProposalSettingsErrorCause` ?